ingestion/src/metadata/ingestion/source/messaging/kinesis/metadata.py

```diff
@@ -64,7 +67,8 @@ def get_topic_list(self) -> Iterable[BrokerTopicDetails]:
                 topics = self.kinesis.list_streams(**args)
                 all_topics.extend(topics["StreamNames"])
                 has_more_topics = topics["HasMoreStreams"]
-                args["ExclusiveStartStreamName"] = all_topics[-1]
+                if len(all_topics) > 0:
+                    args["ExclusiveStartStreamName"] = all_topics[-1]
```

Comment:
nit: probably out of the scope of this PR, but it would be nice if we could use `.get()` to avoid unexpected `KeyError` exceptions. We are doing it in a few places along this file.